pytorch_lightning/utilities/data.py

```diff
@@ -89,17 +89,13 @@ def has_iterable_dataset(dataloader: DataLoader) -> bool:
 
 def has_len(dataloader: Union[DataLoader, Iterable]) -> bool:
     """Checks if a given Dataloader has ``__len__`` method implemented i.e. if it is a finite dataloader or
-    infinite dataloader.
-
-    Raises:
-        ValueError:
-            If the length of Dataloader is 0, as it requires at least one batch
-    """
-
+    infinite dataloader."""
     try:
         # try getting the length
         if len(dataloader) == 0:
-            raise ValueError("`Dataloader` returned 0 length. Please make sure that it returns at least 1 batch")
+            rank_zero_warn(
+                f"`{dataloader.__class__.__name__}` returned 0 length. Please make sure this was your intention."
```

Comment:
I think `dataloader.__class__.__name__` will return `DataLoader` only. You mean `dataloader.dataset` here?

Reply:
This maintains the previous message and follows what's being checked in the code: `len(dataloader)==0`

Reply:
but isn't that upto the dataset to decide?

if not maybe just update to `DataLoader` since `dataloader.__class__.__name__` is static.

Reply:
oh I see.. thanks :)